### PR TITLE
Nicer conditional SQL

### PR DESCRIPTION
**Pull request by mgirlich**
This slightly improves the SQL translations of `is.na()`, `if()`, `ifelse()`, `if_else()` and `case_when()`

Before

```r
library(dbplyr, warn.conflicts = FALSE)
translate_sql(ifelse(is.na(x), 0L, 1L))
#> <SQL> CASE WHEN (((`x`) IS NULL)) THEN (0) WHEN NOT(((`x`) IS NULL)) THEN (1) END

translate_sql(case_when(is.na(x) ~ 0L, TRUE ~ 1L))
#> <SQL> CASE
#> WHEN (((`x`) IS NULL)) THEN (0)
#> ELSE (1)
#> END
```

After

```r
library(dbplyr, warn.conflicts = FALSE)
translate_sql(ifelse(is.na(x), 0L, 1L))
#> <SQL> CASE WHEN ((`x` IS NULL)) THEN 0 WHEN NOT ((`x` IS NULL)) THEN 1 END

translate_sql(case_when(is.na(x) ~ 0L, TRUE ~ 1L))
#> <SQL> CASE WHEN ((`x` IS NULL)) THEN 0 ELSE 1 END
```

### CODE REVIEW

File: R/backend-.R [hunk 1]
```diff
@@ -150,9 +150,15 @@ base_scalar <- sql_translator(
     sql_expr(ROUND(!!x, !!as.integer(digits)))
   },
 
-  `if` = sql_if,
-  if_else = function(condition, true, false, missing = NULL) sql_if(condition, true, false, missing),
-  ifelse = function(test, yes, no) sql_if(test, yes, no),
+  `if` = function(cond, if_true, if_false = NULL) sql_if(
```
Comment (hadley) [inline R/backend-.R:153]: Since these now span multiple lines, I'd add `{}`

File: R/translate-sql-conditional.R [hunk 2]
```diff
@@ -66,5 +81,18 @@ sql_switch <- function(x, ...) {
 }
 
 sql_is_null <- function(x) {
-  sql_expr((((!!x)) %is% NULL))
+  x_expr <- enexpr(x)
+  if (is_call(x_expr)) {
```
Comment (hadley) [inline R/translate-sql-conditional.R:85]: Can you now use `enpar` here?

File: R/translate-sql-conditional.R [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
-      build_sql(" WHEN NOT(", cond, ") THEN (", if_false, ")"),
-    if (!is.null(missing)) {
-      missing_cond <- translate_sql(is.na(!!cond), con = sql_current_con())
-      build_sql(" WHEN ", missing_cond, " THEN (", missing, ")")
-    },
-    " END"
-  )
+  out <- build_sql("CASE WHEN ", enpar(cond), " THEN ", enpar(if_true))
+
+  if (!is.null(if_false) && identical(if_false, missing)) {
```
Comment (hadley) [inline R/translate-sql-conditional.R:4]: I think this deserves a comment to briefly describe the reasoning.

File: tests/testthat/test-translate-sql.R [hunk 4]
```diff
@@ -26,11 +26,11 @@ test_that("between translated to special form (#503)", {
 
 test_that("is.na and is.null are equivalent", {
   # Needs to be wrapped in parens to ensure correct precedence
-  expect_equal(translate_sql(is.na(x)), sql("((`x`) IS NULL)"))
-  expect_equal(translate_sql(is.null(x)), sql("((`x`) IS NULL)"))
+  expect_equal(translate_sql(is.na(x)), sql("(`x` IS NULL)"))
```
Comment (hadley) [inline tests/testthat/test-translate-sql.R:29]: I think this probably needs a new test for `translate_sql(is.na(x + y))` to capture the spirit of the original test.

File: R/translate-sql-conditional.R (hunk 3 above)
Comment (hadley) [reply R/translate-sql-conditional.R:4]: And I don't think I saw a test that covered this branch?

**Review by hadley — APPROVED** [review]

File: R/translate-sql-conditional.R (hunk 2 above)
Comment (mgirlich) [reply R/translate-sql-conditional.R]: Yes, and thanks to that I found an issue with this approach and the pipe

**Comment by mgirlich** [comment]
I had to change from `enexpr()` to `enquo()` due to an issue with the pipe: `translate_sql(x %>% ifelse(1, 2))` didn't work but it does now.

**Review by hadley — APPROVED** [review]

**Pull request merged**